lightning/src/ln/outbound_payment.rs

```diff
@@ -868,6 +869,36 @@ impl OutboundPayments {
 				RetryableSendFailure::DuplicatePayment
 			})?;
 
+		match route_params.payment_params.payee {
+			Payee::Clear { node_id, .. } if node_id == payer_pubkey => {
+				let payment_secret = match recipient_onion.payment_secret {
+					Some(secret) => secret,
+					None => PaymentSecret([0; 32])
+				};
+				let payment_purpose = PaymentPurpose::InvoicePayment {
+					payment_preimage,
+					payment_secret,
+				};
+
+				let payment_preimage = payment_preimage
+			                .unwrap_or_else(|| PaymentPreimage(entropy_source.get_secure_random_bytes()));
+				{
```

Comment:
If I don't take the lock in a block here, whenever I run the test I get the following error: `Tried to violate existing lockorder. Build with the backtrace feature for more info.`. It is coming from `/lightning/src/sync/debug_sync.rs` I couldn't figure out why this error is arising.